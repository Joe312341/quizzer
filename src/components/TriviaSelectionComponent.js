import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, Picker, Slider, Button, StyleSheet } from 'react-native';

//utilities
import { capitalizeString } from '../utilities/stringHelpers';

const TriviaSelectionComponent = ({ difficulty, difficultyLevels, actions, numberOfQuestions }) => (
  <View>
    <Text style={styles.labelText}>Welcome to the Trivia App. Select the difficulty level and the number of questions to start playing!</Text>
    <Picker
      selectedValue={difficulty}
      onValueChange={(newDifficulty) => actions.selectDifficulty(newDifficulty)}
    >
      {difficultyLevels.map((difficultyLevel) => {
        return (
          <Picker.Item key={difficultyLevel} label={capitalizeString(difficultyLevel)} value={difficultyLevel} />
        )
      })}
    </Picker>
    <Text style={styles.labelText}>Select the number of questions</Text>
    <Text style={styles.labelText}>Current Number of questions {numberOfQuestions}</Text>
    <Slider
      value={numberOfQuestions}
      step={1}
      minimumValue={1}
      maximumValue={20}
      onValueChange={(value) => actions.selectNumberOfQuestions(value)}
    />
    <Button onPress={() => actions.requestFetch(difficulty, numberOfQuestions)} title="Start!" />
  </View>
)

const styles = StyleSheet.create({
  difficultyTile: {
    backgroundColor: '#cddc39',
  },
  labelText: {
    fontWeight: '300',
    textAlign: 'center'
  },
})

TriviaSelectionComponent.propTypes = {
  actions: PropTypes.object.isRequired,
  difficulty: PropTypes.string.isRequired,
  numberOfQuestions: PropTypes.number.isRequired,
  difficultyLevels: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default TriviaSelectionComponent;
