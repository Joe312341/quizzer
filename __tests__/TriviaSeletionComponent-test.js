/*global test expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TriviaSelectionComponent from '../src/components/TriviaSelectionComponent';

test('renders initial layout correctly', () => {
  // initial state of reducer
  const difficulty = 'easy';
  const difficultyLevels = ['easy', 'medium', 'hard', 'mix'];
  const numberOfQuestions = 1;
  const actions = {};

  const tree = renderer.create(
    <TriviaSelectionComponent
      actions={actions}
      difficulty={difficulty}
      numberOfQuestions={numberOfQuestions}
      difficultyLevels={difficultyLevels}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders different slider state than initial state correctly', () => {
  // initial state of reducer
  const difficulty = 'medium';
  const difficultyLevels = ['easy', 'medium', 'hard', 'mix'];
  const numberOfQuestions = 1;
  const actions = {};

  const tree = renderer.create(
    <TriviaSelectionComponent
      actions={actions}
      difficulty={difficulty}
      numberOfQuestions={numberOfQuestions}
      difficultyLevels={difficultyLevels}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders different picker state than initial state correctly', () => {
  // initial state of reducer
  const difficulty = 'easy';
  const difficultyLevels = ['easy', 'medium', 'hard', 'mix'];
  const numberOfQuestions = 10;
  const actions = {};

  const tree = renderer.create(
    <TriviaSelectionComponent
      actions={actions}
      difficulty={difficulty}
      numberOfQuestions={numberOfQuestions}
      difficultyLevels={difficultyLevels}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
