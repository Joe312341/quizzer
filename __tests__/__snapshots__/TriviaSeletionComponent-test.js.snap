// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders different picker state than initial state correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Welcome to the Trivia App. Select the difficulty level and the number of questions to start playing!
  </Text>
  <View
    style={undefined}
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Easy",
            "textColor": undefined,
            "value": "easy",
          },
          Object {
            "label": "Medium",
            "textColor": undefined,
            "value": "medium",
          },
          Object {
            "label": "Hard",
            "textColor": undefined,
            "value": "hard",
          },
          Object {
            "label": "Mix",
            "textColor": undefined,
            "value": "mix",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Select the number of questions
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Current Number of questions 
    10
  </Text>
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumValue={20}
    minimumValue={1}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onSlidingComplete={undefined}
    onStartShouldSetResponder={[Function]}
    onValueChange={[Function]}
    step={1}
    style={
      Array [
        Object {
          "height": 40,
        },
        undefined,
      ]
    }
    value={10}
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Start!
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders different slider state than initial state correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Welcome to the Trivia App. Select the difficulty level and the number of questions to start playing!
  </Text>
  <View
    style={undefined}
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Easy",
            "textColor": undefined,
            "value": "easy",
          },
          Object {
            "label": "Medium",
            "textColor": undefined,
            "value": "medium",
          },
          Object {
            "label": "Hard",
            "textColor": undefined,
            "value": "hard",
          },
          Object {
            "label": "Mix",
            "textColor": undefined,
            "value": "mix",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={1}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Select the number of questions
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Current Number of questions 
    1
  </Text>
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumValue={20}
    minimumValue={1}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onSlidingComplete={undefined}
    onStartShouldSetResponder={[Function]}
    onValueChange={[Function]}
    step={1}
    style={
      Array [
        Object {
          "height": 40,
        },
        undefined,
      ]
    }
    value={1}
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Start!
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders initial layout correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Welcome to the Trivia App. Select the difficulty level and the number of questions to start playing!
  </Text>
  <View
    style={undefined}
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Easy",
            "textColor": undefined,
            "value": "easy",
          },
          Object {
            "label": "Medium",
            "textColor": undefined,
            "value": "medium",
          },
          Object {
            "label": "Hard",
            "textColor": undefined,
            "value": "hard",
          },
          Object {
            "label": "Mix",
            "textColor": undefined,
            "value": "mix",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Select the number of questions
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    Current Number of questions 
    1
  </Text>
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumValue={20}
    minimumValue={1}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onSlidingComplete={undefined}
    onStartShouldSetResponder={[Function]}
    onValueChange={[Function]}
    step={1}
    style={
      Array [
        Object {
          "height": 40,
        },
        undefined,
      ]
    }
    value={1}
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Start!
      </Text>
    </View>
  </View>
</View>
`;
